src/bp-forums/admin/forums.php

```diff
@@ -80,6 +80,36 @@ private function setup_actions() {
 
 			// Set forum states
 			add_filter( 'display_post_states', array( $this, 'bbp_set_hidden_forum_states' ), 10, 2 );
+
+			// Filter post parent for forum type post.
+			add_filter( 'wp_insert_post_parent', array( $this, 'filter_forum_parent' ), 10, 4 );
+		}
+
+		/**
+		 * Can update forum parent, If not then return the current parent.
+		 *
+		 * Disable "Forum Parent" when the main Forum associated with Group
+		 *
+		 * @since bbPress (r4067)
+		 *
+		 * @param init   post_parent post parent.
+		 * @param init   post_ID     post ID.
+		 * @param array  new_postarr submited post data.
+		 * @param object postarr     post attributes.
+		 *
+		 * @return init
+		 */
+		function filter_forum_parent( $post_parent, $post_ID, $new_postarr, $postarr ) {
+			if ( $new_postarr['post_type'] != 'forum' ) {
```

Comment:
please use Yoda conditions and check your code in PHPCS as well.

Reply:
#139448c17d98dc5293d88d247b968f9f5052a12d